File: javascript_prime/main_prime.js
function is_prime(num_prime){
    if (isNaN(num_prime)){return "Not a number!"}
    for(let i = 1; i < num_prime; i++) {
        if (num_prime % i === 0 && i !== 1){
            return false;
        }
    }
    return true;
}

function prime_span(num_span){

    for (let i = 1; i < num_span; i++) {
        const num_h = is_prime(i);
        if (num_h == true){
            console.log(i + " is prime!");
        }
    }
}

//console.log(is_prime(13));
const num_interval = 10_0000;
let time_start = Date.now();
prime_span(num_interval);
let time_end = Date.now();
const time_total = time_end - time_start;
console.log("The run time with ", num_interval, " params:");
console.log("in ms: " , time_total, "ms");
console.log("in seconds: ", time_total/1000, "s");

File: lua-prime/main_prime.lua
local function is_prime(number)
    for i = 2, number - 1, 1 do
        if number % i == 0 then
            return false
        end
    end
    return true
end

local function is_prime_range(number)
    for i = 1, number, 1 do
        if is_prime(i) then
           -- print(i," is prime!")
        end

    end
    
end

local time_start = os.clock()
is_prime_range(10000)
local time_end = os.clock()
local time_total = time_end - time_start

print("Ran for ", time_total, " seconds")
print("Ran for ", time_total * 1000, " ms")